### ISSUE TITLE
Multiple hover- classes generated for same css property

### ISSUE THREAD

**Issue by j-panasiuk** [mdgriffith/stylish-elephants#60]
## What I'm trying to do
Capture a few connected style attributes into a list - that list of styles will be a variant for some widget. These can be then easily appended together
```elm
-- define a button variant
success =
    [ Background.color green
    , mouseOver [ Background.color lightGreen ]
    ]

-- default button
button [] { label, onPress }

-- success button
button success { label, onPress }

-- large success button
button (large ++ success) { label, onPress }
```

## What I expected to happen
If same attribute is defined twice on an element, I expected the latter attribute to overwrite the former one. This works as expected most of the time...

## What happens instead
...But weird things happen when the same `mouseOver` attribute (say `Background.color`) is defined twice on an element. In that case each attribute seems to attach its own `hover-background-<value>` class to DOM node. This causes unpredictable behavior, because that node now has 2 competing classes for 1 css property and its style will depend on the order in which these classes will be generated in `<style>` node

It probably sounds a bit convoluted at this point, so here's an exampe in Ellie:
https://ellie-app.com/4Kz9jdjgfa1/0

Bottom line is: elements shouldn't get 2+ competing `hover-background-` classes :)
